### PR TITLE
feat(sensors): Automatic aggregation of more sensor types

### PR DESCRIPTION
**Pull request by Teagan42**
Addresses: #31 

# Description
* Entities extend a common base class -full typesafety is still in place.
* State calculation method is configurable via ConfigFlow
* Foundation for migrating binary sensors to base class is in place
* Sensor types: Humidity and Temperature

### CODE REVIEW

**Review by c-st — APPROVED** [review]
Thank you so much for your PR ! 🚀

While reading through the changes I had a few questions and have added comments for them. I'd test the changes in my local setup and then I don't see anything against creating a new release 👍

**Comment by c-st** [comment]
@Teagan42 I was testing with the local HA setup based on `configuration.yaml` and shuffled a few things around. Hope you don't mind 🙂. If you like, give it a try if everything still works for you, too!

File: custom_components/auto_areas/ha_helpers.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
 def get_all_entities(
     entity_registry: EntityRegistry,
     device_registry: DeviceRegistry,
-    area_id: str,
-    domains: list[str] = None,
+    area_id: str | None,
```
[the remark this answers is not in the document]
Comment (Teagan42) [reply custom_components/auto_areas/ha_helpers.py]: So the return of `area_registry.async_get_area` is possibly `None` - this was to align the type-hints with the warnings I got from my IDE

File: custom_components/auto_areas/auto_entity.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            self.unsubscribe()
+
+    def _get_state(self) -> StateType | None:
+        """Get the state of the sensor."""
+        calculate_state = self.auto_area.get_calculation(self._device_class)
+
+        if calculate_state is None:
+            LOGGER.info("%s unable to get state calculation method",
+                        self._logger_name)
+            return None
```
[the remark this answers is not in the document]
Comment (Teagan42) [reply custom_components/auto_areas/auto_entity.py]: It shouldn't, but I can safely safely say I didn't experience any issues. Not that others might.

File: custom_components/auto_areas/sensor.py [hunk 3]
```diff
@@ -1,126 +1,87 @@
...
+
+    @cached_property
+    @override
+    def native_unit_of_measurement(self) -> str:
+        """Return unit of measurment."""
+        return self._hass.config.units.temperature_unit
+
+
+class HumiditySensor(AutoEntity[SensorEntity, SensorDeviceClass], SensorEntity):
+    """Set up aggregated temperature sensor."""
```
[the remark this answers is not in the document]
Comment (Teagan42) [reply custom_components/auto_areas/sensor.py]: Copy-pasta- missed that one

**Comment by Teagan42** [comment]
I had some issues that needed to be fixed - also added repair flow in case the area is removed/changed

**Comment by c-st** [comment]
Ah good point - that could be a source of all kind of issues. It works really well now. 

Again, thanks a lot for your contribution @Teagan42 ! 🙏

**Pull request merged**